MetaMask/metamask-mobile
# feat: preinstalled mobile snaps - [1 of 10]

## **Description**

Within the scope of [Notifications Project](https://github.com/MetaMask/mobile-planning/issues/1495), and due restrictions on both Apple and Google store, the way Snaps will be delivered to mobile users will be in a pre installed approach. Notifications will rely on [MM auth](https://github.com/consensys-vertical-apps/metamask-auth-sdk/tree/main) 
 as a snap. 

Since preinstalled snaps isn't a  feature available on mobile yet, this PR implements a way to enable MetaMask mobile to read from FS an embedded/preInstalled snap, so we can embed auth snap into Mobile App by default.

## **Related issues**

Fixes: [Jira Ticket](https://consensyssoftware.atlassian.net/browse/NOTIFY-467?atlOrigin=eyJpIjoiNjdiY2QwOGNmZjY3NGQwYjg3YzM3ZTg1NDI1N2RiZWQiLCJwIjoiaiJ9)

## **Manual testing steps**

1. Go to this Snaps page and check the existence of a installed snap in a fresh installation.

## **Screenshots/Recordings**



### **After**

![Screenshot 2024-03-14 at 11 11 47](https://github.com/MetaMask/metamask-mobile/assets/44679989/df684662-8100-4490-9e7c-2a1d0f512dbe)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
